### PR TITLE
feat: add support for resource state events API

### PR DESCRIPTION
**Pull request by nejch**

### CODE REVIEW

File: gitlab/v4/objects/events.py [hunk 1]
```diff
@@ -110,6 +124,18 @@ class ProjectMergeRequestResourceMilestoneEventManager(RetrieveMixin, RESTManage
     _from_parent_attrs = {"project_id": "project_id", "mr_iid": "iid"}
 
 
+class ProjectMergeRequestResourceStateEvent(RESTObject):
+    pass
+
+
+class ProjectMergeRequestResourceStateEventManager(RetrieveMixin, RESTManager):
+    _path = (
+        "/projects/%(project_id)s/merge_requests/%(issue_iid)s/resource_state_events"
```
Comment (max-wittig) [inline gitlab/v4/objects/events.py]: Shouldn't this be 

```
"/projects/%(project_id)s/merge_requests/%(mr_iid)s/resource_state_events"
```

like here? https://github.com/python-gitlab/python-gitlab/pull/1364/commits/40eb59e66ad381bd329a66d9e69db61b10aa5440#diff-9715eb24ecd3f0c1c56e83d4867974145fcffdf50cabb59042173492bb440274R134

File: gitlab/v4/objects/events.py [hunk 2]
```diff
@@ -110,6 +124,18 @@ class ProjectMergeRequestResourceMilestoneEventManager(RetrieveMixin, RESTManage
...
+    )
+    _obj_cls = ProjectMergeRequestResourceStateEvent
+    _from_parent_attrs = {"project_id": "project_id", "issue_iid": "iid"}
```
Comment (max-wittig) [inline gitlab/v4/objects/events.py]: issue_iid -> mr_iid

I don't know how to add suggestions to GitHub anymore.

File: docs/gl_objects/events.rst [hunk 3]
```diff
@@ -2,6 +2,9 @@
 Events
 ######
 
+Events
+======
```
Comment (max-wittig) [inline docs/gl_objects/events.rst:6]: isn't events now duplicated?

**Review by max-wittig — CHANGES_REQUESTED** [review]
Some little typos

File: gitlab/v4/objects/events.py (hunk 1 above)
Comment (nejch) [reply gitlab/v4/objects/events.py]: Copy/paste :facepalm: Done, thanks :grin:

File: docs/gl_objects/events.rst (hunk 3 above)
Comment (nejch) [reply docs/gl_objects/events.rst:6]: Technically yes but without it the next heading isn't displayed with the proper size (see below)

We seem to also do this elsewhere, maybe for the same reason: https://github.com/python-gitlab/python-gitlab/blob/af781c10db3829163f977e494e4008acf2096d64/docs/gl_objects/projects.rst

image

**Pull request merged**